pytype/tests/test_utils.py

```diff
@@ -418,3 +421,35 @@ class Py3Opcodes(object):
   BUILD_CONST_KEY_MAP = 156
   BUILD_STRING = 157
   BUILD_TUPLE_UNPACK_WITH_CALL = 158
+
+
+# We compute the availability of python3.7 with subprocess, which is slow, so
+# the result is cached.
+_IS_37_AVAILABLE = None
+
+
+# pylint: disable=invalid-name
+# Use camel-case to match the unittest.skip* methods.
+# TODO(rechen): Remove skipUnless37Available once python3.7 is available in all
+# of pytype's testing environments.
+def skipUnless37Available(f):
+  """Skip the test unless Python 3.7 is available."""
+  global _IS_37_AVAILABLE
+  if _IS_37_AVAILABLE is None:
+    try:
+      subprocess.call(["python3.7", "-V"])
+    except OSError:
+      _IS_37_AVAILABLE = False
+    else:
+      _IS_37_AVAILABLE = True
+  return unittest.skipUnless(_IS_37_AVAILABLE, "no python3.7")(f)
+
+
+def skipInVersion(python_version, reason):
```

Comment:
we should never need to skip a test in exactly one version, unless that version is (3, 7). so i'd suggest either hardcoding this to skipIn37 or (if you want something more general) changing it to skipInVersionsFrom so that it automatically skips everything >= the supplied version.

Reply:
Changed to skipIn37.